sway/input/text_input.c

```diff
@@ -454,9 +454,6 @@ static void handle_im_popup_surface_unmap(struct wl_listener *listener, void *da
 	struct sway_input_popup *popup =
 		wl_container_of(listener, popup, popup_surface_unmap);
 
-	// relative should already be freed as it should be a child of the just unmapped scene
-	popup->desc.relative = NULL;
-
```

Comment:
This isn't correct, we cannot just remove this because of the mentioned signal order. Both scene and text_input want to do stuff when the surface unmaps. If scene runs before text_input, text_input will be working with an invalid pointer.

Putting `scene_descriptor_destroy()` here like before was correct. Sorry for the confusion.

Reply:
I see, thank you